### DISCUSSION TITLE
Dependencies from other stencil projects

### DISCUSSION THREAD

**Discussion by RuiCardoso021** [stenciljs/core#5886] [Stencil Testing]
I have, two lib [A] and [B]. After install lib [A] into lib [B] and import one component from lib [A] always in build compile components from bouths. My question is, exist any form to compile only internal components and used lib instaled with reference to external component.

Note: I use property "excludeComponents" into angular and react target to exclude this other components but I don't how exclude from stencil lib.

```
import excludeA from '@A/exclude-components';
import { angularOutputTarget} from '@stencil/angular-output-target';
import { Config } from '@stencil/core';
import { reactOutputTarget as react } from '@stencil/react-output-target';
import { sass } from '@stencil/sass';


const excludeComponents = [...excludeA];
export const config: Config = {
  namespace: 'B',
  outputTargets: [
    {
      type: 'dist',
      esmLoaderPath: './loader',
    },
    {
      type: 'dist-custom-elements',
    },
    {
      type: 'docs-readme',
    },
    {
      type: 'www',
      serviceWorker: null, // disable service workers
    },
    react({
      componentCorePackage: 'B',
      loaderDir: 'loader',
      proxiesFile: './../react-B/B/src/components/stencil-generated/index.ts',
      includeDefineCustomElements: true,
      excludeComponents: excludeComponents,
    }),
    angularOutputTarget({
      componentCorePackage: 'B',
      directivesProxyFile: './../angular-B/projects/B/src/libs/stencil-generated/proxies.ts',
      directivesArrayFile: './../angular-B/projects/B/src/libs/stencil-generated/index.ts',
      outputType: 'component',
      excludeComponents: excludeComponents,
    }),
  ],
  plugins: [sass()],
};

```

**Comment by christian-bromann** [comment]
Thanks for raising the discussion @RuiCardoso021. Unfortunately I am not quite sure if I understand this problem correctly. May I ask if you could provide a more detailed description?

**Comment by stout-ni** [comment]
I've encountered this problem before as well. I solved this by removing the `collection` and `collection:main` fields from the package.json file, because Stencil will treat dependencies with those fields as Stencil component libraries and bundle all of their components together.

![image](https://github.com/user-attachments/assets/e7628453-4d83-43e8-85ad-67e3cae7605a)

https://github.com/ionic-team/stencil/blob/main/src/compiler/transformers/collections/add-external-import.ts
